From 7aeb91d12889bd49554d5e2d410b1977a3ba44de Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Tue, 11 Feb 2020 14:34:13 +0100
Subject: [PATCH 259/456] sound: soc: ac100: Make sure we shutdown the audio
 outputs on reboot

There will also be a need to disable ADC clock to avoid ldoin leakage current.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 sound/soc/sunxi/ac100-codec.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/sound/soc/sunxi/ac100-codec.c b/sound/soc/sunxi/ac100-codec.c
index c3fec8d8da52..7d0cdba9c3a9 100644
--- a/sound/soc/sunxi/ac100-codec.c
+++ b/sound/soc/sunxi/ac100-codec.c
@@ -918,6 +918,16 @@ static const struct snd_soc_dapm_route ac100_codec_routes[] = {
 	{ "LINEOUT", NULL, "Line Out" },
 };
 
+static int ac100_codec_set_bias_level(struct snd_soc_component *component,
+				      enum snd_soc_bias_level level)
+{
+	if (level == SND_SOC_BIAS_OFF) {
+		pr_err("XX: idle bias off\n");
+	}
+
+	return 0;
+}
+
 static const struct snd_soc_component_driver ac100_codec_analog_cmpnt_drv = {
 	.controls		= ac100_codec_controls,
 	.num_controls		= ARRAY_SIZE(ac100_codec_controls),
@@ -925,6 +935,7 @@ static const struct snd_soc_component_driver ac100_codec_analog_cmpnt_drv = {
 	.num_dapm_widgets	= ARRAY_SIZE(ac100_codec_widgets),
 	.dapm_routes		= ac100_codec_routes,
 	.num_dapm_routes	= ARRAY_SIZE(ac100_codec_routes),
+	.set_bias_level		= ac100_codec_set_bias_level,
 };
 
 static int ac100_codec_probe(struct platform_device *pdev)
-- 
2.34.1

